### ISSUE TITLE
Warning thrown when binding bar plot - not sure if the warning is expected, and how to resolve if so

### ISSUE THREAD

**Issue by vsriram11** [holoviz/panel#6451]
When I run the below -

```py3
select1 = pn.widgets.MultiSelect(...)
select2 = pn.widgets.MultiSelect(...)
select3 = pn.widgets.MultiSelect(...)

def create_plot(p, i, g):
    dff = df[df['id1'].isin(p) & df['id2'].isin(i) & df['id3'].isin(g)].groupby(['id3'])['display1', 'display2', 'display3'].mean()
    return dff.hvplot.bar(width=1000, xlabel='x', ylabel="y", rot=90)

bound_plot = pn.bind(create_plot, p=select1, i=select2, g=select3)
app = pn.Column(select1, select2, select3, bound_plot)
app.servable()
```

I would expect this to work without warnings, but instead I get the following warning:

`Parameter 'object' on <class 'panel.pane.base.PaneBase'> is being given a valid parameter reference <function _param_bind.<locals>.wrapped at 0x7f72fbcdfaf0> but is implicitly allow_refs=False. In future allow_refs will be enabled by default and the reference <function _param_bind.<locals>.wrapped at 0x7f72fbcdfaf0> will be resolved to its underlying value :Bars   [id3,Variable]   (value). Please explicitly set allow_ref on the Parameter definition to declare whether references should be resolved or not.`

I'm not sure if this is expected behavior, and if so, how might I go about resolving this? Apologies if this turns out to be more of a query than an actual bug.

**Comment by philippjfr** [comment]
No, this warning isn't your fault really. Which version of Panel are you running?

**Comment by vsriram11** [comment]
@philippjfr thanks for responding! I'm using version panel==1.2.3, which is the last version of panel supporting python 3.8 (need this because the env I'm working in requires python 3.8).

**Comment by svew** [comment]
I'm experiencing the same issue with a simpler binding, is there a fix for this?

**Comment by nanodust** [comment]
I also had this issue with holoviews 1.17.1, panel 1.2.3, and python 3.8.20. 

The result was that initial plot worked - yet in any 'binding' demo (e.g. [crossfilter](https://holoviews.org/gallery/apps/bokeh/crossfilter.html)) the GUI never 'binds' - none of the combo boxes update the plot.  and I received this warning. 

it stumped me for a while, could not find a resolution until I just upgraded Python, and now the binding demos work well. 

python 3.10.12, holoviews 1.17.1, and panel version 0.14.4.

yet in trying on another environment, the binding does not work, *and* we get no warning or error. 

python 3.10.12, holoviews 1.20.0, and panel version 1.5.3 

so it's a bit of version roulette. 

Happy it's working, yet I shall look for some guidance for tagged versions that are tested and compatible before I depend on Holoviews instead of just bokeh/matplotlib/plotly directly.
